test/SourceKit/NameTranslation/basic.swift

```diff
@@ -66,6 +67,7 @@ class MyDerived: FooClassDerived {
 // CHECK9: fooProperty11
 // CHECK10: init(nstanceFunc21:)
 // CHECK11: init(lassDerived2:)
+// CHECK11-1: FooClassDerived2
```

Comment:
Renaming from the `Foo` in `Foo(x: 1)` will currently rename the typename, which tries to name-translate the initializer name, leading to broken code after rename.  If possible, it would be best to have this location work for both init and typename in both directions.